src/wallet-account.ts

```diff
@@ -169,6 +179,51 @@ export class WalletConnection {
         window.location.assign(newUrl.toString());
     }
 
+    /**
+     * Allows the user to send a batch of transactions to one or multiple recievers with only one redirect to the NEAR wallet.
+     * @param transactionOptions Array of Transaction options that will be requested to sign
+     * @param callbackUrl The url to navigate to after the user is prompted to sign
+     * 
+     * @example
+     * ```js
+     * async function run() {
+     *      const near = await connect({
+     *          keyStore: new keyStores.BrowserLocalStorageKeyStore(),
+     *          networkId: "testnet",
+     *          walletUrl: 'https://wallet.testnet.near.org',
+     *          nodeUrl: "https://rpc.testnet.near.org",
+     *      })
+     *      
+     *      const wallet = new WalletConnection(near);
+     *      wallet.batchSendTransactions([
+     *          { receiverId: 'contract-1.testnet', actions: [functionCall('contractMethod', { arg: 'value' }, new BN('30000000000000'), new BN(utils.format.parseNearAmount('1')))] },
+     *          { receiverId: 'user-account-1.testnet', actions: [transfer(new BN(utils.format.parseNearAmount('1'))), transfer(new BN(utils.format.parseNearAmount('4')))] }
+     *      ]);
+     * }
+     * ```
+     */
+    async batchSendTransactions(transactionOptions: TransactionOptions[], callbackUrl?: string) {
+        const accountId = this.getAccountId();
+        const localKey = await this._near.connection.signer.getPublicKey(accountId, this._near.connection.networkId);
+        
+        const block = await this._near.connection.provider.block({ finality: 'final' });
+        const blockHash = baseDecode(block.header.hash);
+                            
+        const transactions: Transaction[] = await Promise.all(transactionOptions.map(async({ receiverId, actions }, index) => {
+            // TODO: Cache & listen for nonce updates for given access key
+            const accessKey = await this.account().accessKeyForTransaction(receiverId, actions, localKey);
+            if (!accessKey) {
+                throw new Error(`Cannot find matching key for transaction sent to ${receiverId}`);
+            }
+            const publicKey = PublicKey.from(accessKey.public_key);
+        
+            const nonce = accessKey.access_key.nonce + index + 1;
```

Comment:
we actually need to increment the nonce here instead of using the index. Maybe:

```js
const nonce = ++accessKey.access_key.nonce
```

I wonder if we actually want a function to do this? like:

```js
const nonce = accessKey.access_key.nextNonce();
```

Reply:
Yeah I see what you mean. The first one would definitely work